lib/Evaluate/characteristics.cpp

```diff
@@ -1,21 +1,21 @@
-//===-- lib/evaluate/characteristics.cpp ----------------------------------===//
+//===-- lib/Evaluate/characteristics.cpp ----------------------------------===//
 //
 // Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
 
-#include "flang/evaluate/characteristics.h"
-#include "flang/common/indirection.h"
-#include "flang/evaluate/check-expression.h"
-#include "flang/evaluate/fold.h"
-#include "flang/evaluate/intrinsics.h"
-#include "flang/evaluate/tools.h"
-#include "flang/evaluate/type.h"
-#include "flang/parser/message.h"
-#include "flang/semantics/scope.h"
-#include "flang/semantics/symbol.h"
+#include "flang/Evaluate/characteristics.h"
+#include "flang/Common/indirection.h"
```

Comment:
These two are the wrong way around. Could fix at the same time?

Reply:
This behaviour is really odd. 
The problem I believe could come from the git-rebase. 
Because this does not make any sense to replace totally different names.

Reply:
@RichBarton-Arm  which files?
#include "flang/evaluate/characteristics.h" .    #include "flang/Evaluate/characteristics.h"$
#include "flang/common/indirection.h" .           #include "flang/Common/indirection.h"
#include "flang/evaluate/check-expression.h" . #include "flang/Evaluate/check-expression.h"
#include "flang/evaluate/fold.h" .                       #include "flang/Evaluate/fold.h"
#include "flang/evaluate/intrinsics.h"                   #include "flang/Evaluate/intrinsics.h"
#include "flang/evaluate/tools.h"                      #include "flang/Evaluate/tools.h"
#include "flang/evaluate/type.h"                    #include "flang/Evaluate/type.h"
#include "flang/parser/message.h"                #include "flang/Parser/message.h"
#include "flang/semantics/scope.h"               #include "flang/Semantics/scope.h"
#include "flang/semantics/symbol.h"              #include "flang/Semantics/symbol.h"

Reply:
Sorry - I mean that the rest of the includes seem to be in alphabetical order except for "flang/Common/indirection.h", which looks like it should be above "flang/Evaluate/characteristics.h". Clearly a bug in the original code though, so happy for this patch not to address it.

Reply:
This should be caught by clang-format but in some places clang-format hasn't fully been run on every file. This should be fixed in the clang-format alignment patch at #945

Reply:
characteristics.h is first because this is characteristics.cpp, so the order is right.

Reply:
Ah I didn't spot that. Yes having actually looked at it rather than just reading the comments the order looks correct to me too.